spec.md

```diff
@@ -2734,6 +2779,301 @@ message NodeExpandVolumeResponse {
+
+#### `GroupControllerGetCapabilities`
+
+A GroupController Plugin MUST implement this RPC call. This RPC allows the CO to check the supported capabilities of group controller service provided by the Plugin.
+
+```protobuf
+message GroupControllerGetCapabilitiesRequest {
+  // Intentionally empty.
+}
+
+message GroupControllerGetCapabilitiesResponse {
+  // All the capabilities that the group controller service supports.
+  // This field is OPTIONAL.
+  repeated GroupControllerServiceCapability capabilities = 1;
+}
+
+// Specifies a capability of the group controller service.
+message GroupControllerServiceCapability {
+  message RPC {
+    enum Type {
+      UNKNOWN = 0;
+
+      // Indicates that the group controller plugin supports
+      // creating and deleting a volume group snapshot.
+      CREATE_DELETE_VOLUME_GROUP_SNAPSHOT = 1
+      [(alpha_enum_value) = true];
+
+      // Indicates whether the group controller plugin supports
+      // creating a volume from an individual volume snapshot if the
+      // volume snapshot is part of a VolumeGroupSnapshot.
+      // Use cases: selective restore, advanced recovery, etc.
+      INDIVIDUAL_SNAPSHOT_RESTORE = 2 [(alpha_enum_value) = true];
+
+      // Indicates that the group controller plugin supports getting
+      // details of a volume group snapshot.
+      GET_VOLUME_GROUP_SNAPSHOT = 3 [(alpha_enum_value) = true];
+
+      // Indicates that the group controller plugin supports getting
+      // details of a list of volume group snapshots.
+      LIST_VOLUME_GROUP_SNAPSHOTS = 4 [(alpha_enum_value) = true];
+    }
+
+    Type type = 1;
+  }
+
+  oneof type {
+    // RPC that the controller supports.
+    RPC rpc = 1;
+  }
+}
+```
+
+##### ControllerGetCapabilities Errors
+
+If the plugin is unable to complete the ControllerGetCapabilities call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+
+#### `CreateVolumeGroupSnapshot`
+
+**ALPHA FEATURE**
+
+A Group Controller Plugin MUST implement this RPC call if it has `CREATE_DELETE_VOLUME_GROUP_SNAPSHOT` controller capability.
+This RPC will be called by the CO to create a new volume group snapshot from a list of source volumes on behalf of a user.
+
+The purpose of this call is to request the creation of a multi-volume snapshot. Group snapshots can be created from the existing list of volumes. Note that calls to this function MUST be idempotent - the function may be called multiple times for the same name - the group snapshot must only be created once.
+
+In VolumeGroupSnapshot message, snapshots is an optional field while group_snapshot_id is a required field.
+
+```protobuf
+message CreateVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // suggested name for a group snapshot (required for idempotent)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // volume ids of the volumes to be snapshotted together.
+  // This field is REQUIRED.
+  repeated string volume_ids = 2;
```

Comment:
In `CreateSnapshotRequest`, this is called `source_volume_id`. Do we want to be consistent here?